llm-chain-openai/src/chatgpt/text_splitter.rs

```diff
@@ -0,0 +1,122 @@
+use super::step::Model;
+use llm_chain::text_splitter::{TextSplitError, TextSplitter};
+use tiktoken_rs::CoreBPE;
+
+pub struct OpenAITextSplitter {
+    model: Model,
+}
+
+impl OpenAITextSplitter {
```

Comment:
If you could make: 

https://github.com/sobelio/llm-chain/blob/09f698af41be963d1c3d4f481a1ef656c5f1ac9d/llm-chain-openai/src/chatgpt/executor.rs#L71-L82

Use this system that would be great :)